Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -970,6 +975,12 @@ private void UpdatePointer(PointerData pointerData)
                     hitResult3d.Clear();
                     QueryScene(pointerData.Pointer, raycastProvider, prioritizedLayerMasks, hitResult3d, maxQuerySceneResults, focusIndividualCompoundCollider);
 
+                    if ((gazeProviderPointingData?.Pointer != null) &&
```

Comment:
Do you have some more details on why Pointer can be null now but before it wouldn't be?

i.e. in general adding more null checks for pointers is a bit scary because IMixedRealityPointer is implemented by classes that extend MonoBehaviour, and are thus a bit dangerous to do null checks against:

One such path:
IMixedRealityPointer -> TouchPointer -> BaseControllerPointer -> ControllerPoseSynchronizer -> InputSystemGlobalHandlerListener -> MonoBehaviour

I know we do this already in other places, I think we have to be careful in how we choose to add them going forward.

Reply:
It always could have been null. We were just never handling the case.

Essentially, it _will_ be null for at least one frame after changing profiles.

Reply:
The null check prevents an exception and allows later code to properly execute.